### ISSUE TITLE
Further clarification of CC-BY-SA 3.0 Licenses

### ISSUE THREAD

**Issue by reknih** [citation-style-language/styles#6751]
This issue is related to #4129. We want to include CSL citation styles in Typst. We plan to convert them to CBOR, a binary data format with [citationberg](https://github.com/typst/citationberg) and then included in some or all distributions of the Typst binary.

I have the following questions with regard to the interpretation of the terms "collective work" and "derivative work" with regards to CSL under the CC-BY-SA 3.0 licenses:
- It is clear that we must share the CBOR dumps under CC-BY-SA 3.0. Must the software producing them be available under the same license?
- #4129 states that CSL files may be distributed alongside a binary without requiring the binary to be available under CC-BY-SA 3.0. Can we link the files into the same binary as the [Typst compiler](https://github.com/typst/typst) without requiring Typst to be available under CC-BY-SA 3.0 in its entirety?

**Comment by adam3smith** [comment]
tldr; : no and yes

Long version: I'm still not sure if CC-BY-SA was the right license choice (but it's about 1000 contributors too late to change now), but the purpose of the license was never any strong virality. All we want is for people to
1.) Attribute the CSL project as a whole
2.) Not distribute the styles without the embedded author metadata
3.) Not distribute the styles as a proprietary package by themselves

In addition, for your particular case and with Typst and citationberg both being open source , because CC-BY-SA is not a very common (or, for that matter, very good) license for software, we'd interpret the "share alike" stipulation very broadly to encompass  [any license categorized as Open Source by OSI](https://opensource.org/licenses/) as meeting "Share alike" as long as it's clear that the application of that license to a project that contains CSL styles doesn't actually contsitute a re-licensing of the CSL styles themselves.

**Comment by adam3smith** [comment]
Please close here if that sufficiently clarifies this for your purposes

**Comment by reknih** [comment]
Thank you!
